src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,50 @@
+package duke.parser;
+
+import duke.command.ByeCommand;
+import duke.command.CheckCommand;
+import duke.command.Command;
+import duke.command.DeadlineCommand;
+import duke.command.DeleteCommand;
+import duke.command.EventCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.command.TodoCommand;
+import duke.command.UnmarkCommand;
+import duke.dukeexception.DukeInvalidCommand;
+
+public class Parser {
+    /**
+     * Returns the corresponding command given the input string.
+     * @param fullCommand the String command to be parsed
+     * @return the command corresponding to the string
+     * @throws DukeInvalidCommand if string is not a valid command
+     */
+    public static Command parse(String fullCommand) throws DukeInvalidCommand {
+        String[] input = fullCommand.split(" ", 2);
+        String command = input[0];
+        if (command.equals("bye")) {
```

Comment:
I like that the if else statements are formatted nicely, but I would also like to suggest switch case as a slightly less verbose alternative.